File: todolist/src/types/memory.ts
export interface Memory {
  id: string;
  date: string;
  content: string;
  context: string;
  created_at: string;
  updated_at: string;
}

export interface CreateMemoryParams {
  date: string;
  content: string;
  context: string;
}

export interface UpdateMemoryParams {
  content?: string;
  context?: string;
}

File: todolist/src-tauri/src/models/memory.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Memory {
    pub id: String,
    pub date: String,
    pub content: String,
    pub context: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateMemoryParams {
    pub date: String,
    pub content: String,
    pub context: String,
}

#[derive(Debug, Deserialize)]
pub struct UpdateMemoryParams {
    pub content: Option<String>,
    pub context: Option<String>,
}
